components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,254 @@
+    // doesn't happen until a batch at the end
+
+    setShouldDisplayResults(true)
+    if (!isFilteringOrgs && isFilteringBills) {
+      return setShouldDisplayResults(false)
+    } else if (!isFilteringOrgs) {
+      return setNotificationsDisplay(billResults)
+    } else if (isFilteringBills) {
+      return setNotificationsDisplay(orgResults)
+    }
+    return setNotificationsDisplay(allResults)
+  }
+
+  const onBillFilterChange = (isFiltering: boolean) => {
+    const filterUpdatingTo = !isFiltering
+    setIsFilteringBills(filterUpdatingTo)
+
+    // isFilteringBills has to be inverted because setState (i.e. setIsFilteringBills)
+    // doesn't happen until a batch at the end
+
+    setShouldDisplayResults(true)
+    if (isFilteringOrgs && !isFilteringBills) {
+      return setShouldDisplayResults(false)
+    } else if (isFilteringOrgs) {
+      return setNotificationsDisplay(billResults)
+    } else if (!isFilteringBills) {
+      return setNotificationsDisplay(orgResults)
+    }
+    return setNotificationsDisplay(allResults)
+  }
+
+  const notificationQuery = async () => {
+    let notificationList: Notifications = []
+    const q = query(subscriptionRef, where("uid", "==", `${uid}`))
+    const querySnapshot = await getDocs(q)
+    querySnapshot.forEach(doc => {
+      // doc.data() is never undefined for query doc snapshots
+      notificationList.push(doc.data().notification)
+    })
+
+    if (notificationsDisplay.length === 0 && notificationList.length != 0) {
+      setAllResults(notificationList)
+      setOrgResults(
+        notificationList.filter(notification => notification.type === "org")
+      )
+      setBillResults(
+        notificationList.filter(notification => notification.type === "bill")
+      )
+      setNotificationsDisplay(allResults)
+    }
+  }
+
+  useEffect(() => {
+    uid ? notificationQuery() : null
+  })
+
+  function Filters() {
+    return (
+      <FilterBoxes
+        isMobile={isMobile}
+        onOrgFilterChange={(isFiltering: boolean) => {
+          onOrgFilterChange(isFiltering)
+        }}
+        onBillFilterChange={(isFiltering: boolean) => {
+          onBillFilterChange(isFiltering)
+        }}
+        isFilteringOrgs={isFilteringOrgs}
+        isFilteringBills={isFilteringBills}
+      />
+    )
+  }
+
+  return (
+    <>
+      {loading ? (
+        <Row>
+          <Spinner animation="border" className="mx-auto" />
+        </Row>
+      ) : (
+        <>
+          {profile ? (
+            <>
+              <StyledContainer>
+                <Header>
+                  <HeaderTitle>Newsfeed</HeaderTitle>
+                  {isMobile ? (
+                    <Filters />
+                  ) : (
+                    <Col className="d-flex flex-column">
+                      <Filters />
+                    </Col>
+                  )}
+                </Header>
+                {shouldDisplayResults ? (
```

Comment:
Instead of keeping a separate state of `shouldDisplayResults`, we might be able to just check the results array itself to see if it has elements that can be displayed.

This reduces some unnecessary state, as well as makes it clear that `filteredResults` determines the rendering of the elements below.

```suggestion
                {filteredResults.length > 0 ? (
```